### PR TITLE
chore(deps): update Node to v12

### PR DESCRIPTION
**Pull request by eunjae-lee**
**Summary**

1. `examples/react-native` and `examples/react-native-query-suggestions` has been re-written. I couldn't update their dependencies correctly to make them work. So I created new projects from scratch by using `expo`, and I copied&pasted the same code we had there.

2. The only part I changed in the examples is, replacing `ListView` to `FlatList`. `ListView` is deprecated, so I had to do that.

3. Except for the examples, there isn't much. Let me know if something isn't clear.

And let me know if you wish me to tidyng up this PR by creating stacked PRs for those examples.

### CODE REVIEW

**Comment by eunjae-lee** [comment]
There was one part that caused many failing tests.
I haven't digged enough but, I guess it has something to do with the change of implementation of Array.sort().

https://github.com/nodejs/node/issues/24294

I've added the commit in this PR: 0934786
Do you think it's okay or should I create another PR for visibility?

**Comment by eunjae-lee** [comment]
I didn't manage to run `react-native` and `react-native-query-suggestions` examples with Node v12.
Many dependencies were outdated. After many trials, I just deleted the examples and re-created them with `expo init <project-name>` and copy&paste the contents into the new examples.
This way, we kind of lose the history in the examples, but there aren't much history to track, IMO.

**Comment by eunjae-lee** [comment]
The only remaining part in this PR is to make `test:integration` work.
For some reason, it's failing.
So I checked out `master` branch on my computer and it still does not work.

Does someone, probably @yannickcr, know if there's any special config to run this locally on `master` branch?

```
Screenshots in progress...
(node:6193) UnhandledPromiseRejectionWarning: Error: net::ERR_EMPTY_RESPONSE at http://localhost:8080/iframe.html?id=integration-with-other-libraries--airbnb-rheostat
    at navigate (/Users/eunjaelee/workspace/react-instantsearch/node_modules/puppeteer/lib/FrameManager.js:120:37)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:189:7)
  -- ASYNC --
    at Frame.<anonymous> (/Users/eunjaelee/workspace/react-instantsearch/node_modules/puppeteer/lib/helper.js:111:15)
    at Page.goto (/Users/eunjaelee/workspace/react-instantsearch/node_modules/puppeteer/lib/Page.js:674:49)
    at Page.<anonymous> (/Users/eunjaelee/workspace/react-instantsearch/node_modules/puppeteer/lib/helper.js:112:23)
    at takeScreenshot (/Users/eunjaelee/workspace/react-instantsearch/integration/takeScreenshot.js:14:14)
    at /Users/eunjaelee/workspace/react-instantsearch/integration/runTest.js:46:11
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:189:7)
(node:6193) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:6193) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

File: .circleci/config.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ aliases:
 defaults: &defaults
   working_directory: ~/react-instantsearch
   docker:
-    - image: circleci/node:8.16.1@sha256:9e83d4b3b046d1c7591952865bc4d803d2ddeaf1c6352f76c2133bbe96606d40
+    - image: circleci/node:12.14.1@sha256:bc28f8653824cc4e28902e73d140011c94dfeda8b1f83393d049daa468f51693
```
Comment (Haroenv) [inline .circleci/config.yml:26]: This no longer has -browsers. Expected?

File: packages/react-instantsearch-core/src/core/createInstantSearchManager.js [hunk 2]
```diff
@@ -29,10 +29,13 @@ const isIndexWidgetEqualIndex = (widget, indexId) =>
   widget.props.indexId === indexId;
 
 const sortIndexWidgetsFirst = (firstWidget, secondWidget) => {
-  if (isIndexWidget(firstWidget)) {
+  const isFirstWidgetIndex = isIndexWidget(firstWidget);
+  const isSecondWidgetIndex = isIndexWidget(secondWidget);
+
+  if (isFirstWidgetIndex && !isSecondWidgetIndex) {
```
Comment (Haroenv) [inline packages/react-instantsearch-core/src/core/createInstantSearchManager.js:35]: Why did you need to make these changes?

File: .circleci/config.yml (hunk 1 above)
Comment (eunjae-lee) [reply .circleci/config.yml:26]: From my trials, only the integration test requires `-browser` image.

File: packages/react-instantsearch-core/src/core/createInstantSearchManager.js (hunk 2 above)
Comment (eunjae-lee) [reply packages/react-instantsearch-core/src/core/createInstantSearchManager.js:35]: Regarding that function, lots of tests failed. I searched for it and there has been a change in the implementation of Array.sort() in Node v11 ([link](https://github.com/nodejs/node/issues/24294)).
And the previous implementation of `sortIndexWidgetsFirst` didn't consider the both widgets being index. After fixing them like this, all the tests passed.

File: .circleci/config.yml [hunk 3]
```diff
@@ -81,7 +81,7 @@ jobs:
   test_integration:
     <<: *defaults
     docker:
-      - image: circleci/node:8.16.1-browsers@sha256:e2f01766ecc1e133ca4873b7a0fa292da2d39e756783f6a7f90d4d5c2a59ccb7
+      - image: circleci/node:12.14.1-browsers
```
Comment (yannickcr) [inline .circleci/config.yml:84]: You could keep the sha256 just to be sure to always pull the same image:

```suggestion
      - image: circleci/node:12.14.1-browsers@sha256:e50f174a64b0cc2622bd9c3202ef2e1c3495ff281ed653da344350745ac606a2
```

Comment (eunjae-lee) [reply .circleci/config.yml:84]: Oh I didn't know it was for that. Thanks 👍
